test/hotspot/jtreg/compiler/lib/generators/Generators.java

```diff
@@ -0,0 +1,472 @@
+     * The result is an unrestrictable generator. If you want a restrictable generator that selects values from a
+     * list and are working with Comparable values, use {@link #orderedRandomElement(Collection)}.
+     */
+    public <T> Generator<T> randomElement(Collection<T> list) {
+        return new RandomElementGenerator<>(this, list);
+    }
+
+    /**
+     * Returns a restrictable generator that uniformly randomly samples elements from the provided collection.
+     * Duplicate elements are discarded from the collection.
+     */
+    public <T extends Comparable<T>> RestrictableGenerator<T> orderedRandomElement(Collection<T> list) {
+        NavigableSet<T> set = list instanceof NavigableSet<T> ? (NavigableSet<T>) list : new TreeSet<>(list);
+        return new RestrictableRandomElementGenerator<>(this, set);
+    }
+
+    /**
+     * Returns a generator that always generate the provided value.
+     */
+    public <T> Generator<T> single(T value) {
+        return new SingleValueGenerator<>(value);
+    }
+
+    /**
+     * Returns a restrictable generator that always generate the provided value.
+     */
+    public <T extends Comparable<T>> RestrictableGenerator<T> single(T value) {
+        return new RestrictableSingleValueGenerator<>(value);
+    }
+
+    /**
+     * Returns a new generator that samples its next element from either generator A or B, with assignable weights.
+     */
+    public <T> Generator<T> mixed(Generator<T> a, Generator<T> b, int weightA, int weightB) {
+        return new MixedGenerator<>(this, a, b, weightA, weightB);
+    }
+
+    /**
+     * Randomly pick an int generator.
+     *
+     * @return Random int generator.
+     */
+    public Generator<Integer> ints() {
+        switch(random.nextInt(0, 6)) {
+            case 0  -> { return uniformInts(); }
+            case 1  -> { return specialInts(0); }
+            case 2  -> { return specialInts(2); }
+            case 3  -> { return specialInts(16); }
+            case 4  -> { return mixedWithSpecialInts(1, 1, 16); }
+            case 5  -> { return mixedWithSpecialInts(1, 2, 2); }
+            default -> { throw new RuntimeException("impossible"); }
+        }
+    }
+
+    /**
+     * A generator of special ints. Special ints are powers of two or values close to powers of 2, where a value
+     * is close to a power of two p if it is in the interval [p - range, p + range]. Note that we also consider negative
+     * values as powers of two.
+     */
+    public RestrictableGenerator<Integer> specialInts(int range) {
+        TreeSet<Integer> set = new TreeSet<>();
+        for (int i = 0; i < 32; i++) {
+            int pow2 = 1 << i;
+            for (int j = -range; j <= range; j++) {
+                set.add(+pow2 + j);
+                set.add(-pow2 + j);
+            }
+        }
+        return orderedRandomElement(set);
+    }
+
+    public Generator<Integer> mixedWithSpecialInts(int weightA, int weightB, int rangeSpecial) {
```

Comment:
```suggestion
    public Generator<Integer> uniformMixedWithSpecialInts(int weightA, int weightB, int rangeSpecial) {
```
Optional, you can also leave it as is.

Reply:
Oh, yeah, agreed. I wanted to change this and then forgot.